src/main/java/com/iexec/common/chain/ChainDeal.java

```diff
@@ -48,16 +48,20 @@
     BigInteger workerStake;
     BigInteger schedulerRewardRatio;
 
-    public static List<String> stringParamsToList(String params) {
-        List<String> listParams;
+    public static DealParams stringToDealParams(String params) {
         try {
-            LinkedHashMap tasksParamsMap = new ObjectMapper().readValue(params, LinkedHashMap.class);
-            listParams = new ArrayList<String>(tasksParamsMap.values());
+            DealParams dealParams = new ObjectMapper().readValue(params, DealParams.class);
+            if(dealParams.getIexec_input_files() == null) {
+                dealParams.setIexec_input_files(new ArrayList<>());
+            }
+            return dealParams;
         } catch (IOException e) {
             log.warn("Params string is not a JSON, considering the string is one full param");
```

Comment:
You can remove this log (to many occurrences in prod)

Reply:
done